pyccel/codegen/printing/ccode.py

```diff
@@ -586,6 +602,68 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        base = self._print(expr.base.label)
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = Shape(expr.base)
+        allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
+                expr.base.internal_variable.allows_negative_indexes)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], Integer):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[i]
+                    inds[i] = Slice.__new__(Slice, start, end)
+                #indices of indexedElement of len==1 shouldn't be a Tuple
+                if isinstance(ind, Tuple) and len(ind) == 1:
+                    inds[i].args = ind[0]
+                if allow_negative_indexes and not isinstance(ind, Integer):
+                    inds[i] = PyccelMod(ind, base_shape[i])
+        inds = [self._print(i) for i in inds]
+        #set dtype to the C struct types
+        dtype = self.find_in_dtype_registry(format(expr.dtype), expr.precision, array=True)
```

Comment:
What is the `format` function doing here?

Reply:
i needed it to return a string i will use self._print i think its better

Reply:
I don't think the `format` function is used anywhere in our code (although we often use `str.format`). I used ipython to get the docstring:
```
Signature: format(value, format_spec='', /)
Docstring:
Return value.__format__(format_spec)

format_spec defaults to the empty string.
See the Format Specification Mini-Language section of help('FORMATTING') for
details.
Type:      builtin_function_or_method
```
I have no idea what happens if you try to specify a format when you call this function on `expr.dtype`. If you don't specify a format then it defaults to `str(expr.dtype)`. If you just want to return a string, this is usually a clear way of doing that. Calling `str(a)` calls the method `a.__str__`. This method is implemented for datatypes and *should* return the same thing as `self._print(expr.dtype)` however there is nothing forcing these 2 to return the same thing. It is therefore better to use the same convention as in the rest of the code to be consistent and reduce the chance of breaking the behaviour in the future if we decide to add more information to the `DataType.__str__` method to help with debugging. In the rest of the code `str(expr.dtype)` is used for printing debugging methods and `self._print(expr.dtype)` is passed to the `self.find_in_dtype_registry` method

Reply:
thank for explaining i will change it